ledger/src/blockstore_db.rs

```diff
@@ -330,7 +333,13 @@ impl Rocks {
                 oldest_slot,
             ),
             AccessType::TryPrimaryThenSecondary => {
-                match DB::open_cf_descriptors(&db_options, path, cfs) {
+                let error_if_log_file_exist = false;
+                match DB::open_cf_for_read_only(
```

Comment:
I'm currently of the opinion that we should disable compactions for the `TryPrimaryThenSecondary` case. Unfortunately, `open_cf_for_read_only()` and `open_cf_as_secondary()` both take CF names instead of descriptors (descriptor is basically a struct for name and options), which means we don't pass in our options and the defaults (automatic compactions enabled by default) are used.

I currently have a PR upstream to add methods such that we could pass descriptors for read_only and secondary cases